lmcache/storage_backend/local_backend.py

```diff
@@ -563,3 +564,211 @@ def close(self):
+            logger.debug(
+                f"Betched Read Signal for {len(self.written_key)} keys:")
+            # print("Write Ready",self.written_key,self.written_size)
+            signal = self.proxy.batched_write_ready(self.written_key,
+                                                    self.written_size)
+            if not signal:
+                logger.error("Write Ready Signal Failed")
+            self.update_lock.release()
+            self.written_key = []
+            self.written_size = []
+
+    @_lmcache_nvtx_annotate
+    @torch.inference_mode()
+    def put_nonblocking(
+        self,
+        key: CacheEngineKey,
+        kv_chunk: torch.Tensor,
+    ) -> None:
+        path = self.cached_paths[key]
+        del self.cached_paths[key]
+        # Abort put if cache too big
+        if path == "" or path is None:
+            return
+
+        kv_obj = None
+
+        # Allocate the kv chunk
+        while kv_obj is None:
+            self.update_lock.acquire()
+            kv_obj = self.cpu_mbufferpool.allocate(kv_chunk)
+            self.update_lock.release()
+            if kv_obj is None:
+                # TODO(Jiayi): Please tune the sleep time for better performance
+                time.sleep(0.01)
+
+        put_stream = torch.cuda.Stream()
+        put_stream.wait_stream(torch.cuda.default_stream(kv_chunk.device))
+        with torch.cuda.stream(put_stream):
+            kv_obj.data.copy_(kv_chunk, non_blocking=True)
+            kv_chunk.record_stream(put_stream)
+        put_stream.synchronize()
+
+        self.remain_writing = self.remain_writing + 1
+        future = self.proc_pool_executor.submit(save_disk_with_callback, path,
+                                                kv_obj, key)
+        future.add_done_callback(self.save_end_callback)
+
+    @_lmcache_nvtx_annotate
+    @torch.inference_mode()
+    def put_blocking(
+        self,
+        key: CacheEngineKey,
+        kv_chunk: torch.Tensor,
+    ) -> None:
+        path = self.cached_paths[key]
+        del self.cached_paths[key]
+
+        if path == "":
+            return
+        logger.debug(f"Saving cache to {path}")
+
+        self.update_lock.acquire()
+        save_file({"kv_chunk": kv_chunk}, path)
+        self.update_lock.release()
+
+        self.proxy.write_ready(key.to_string(), _get_size_in_gb(kv_chunk))
+
+    @_lmcache_nvtx_annotate
+    def get(
+        self,
+        key: CacheEngineKey,
+    ) -> Optional[KVCache]:  # type: ignore
+        """
+        Retrieve the KV cache chunk by the given key
+
+        Input:
+            key: the key of the token chunk, including prefix hash and format
+        Output:
+            the kv cache of the token chunk, in the format of nested tuples
+            None if the key is not found
+        """
+        path = self.proxy.read_check(key.to_string())  # type: ignore
+
+        if path == "":
+            return None
+
+        self.update_lock.acquire()
```

Comment:
Why lock here in `get` but no lock in the following `batched_get` function?

Reply:
This conflict will be handled by address manager.